mantidproject/mantid
# Add progress bars to calling algorithms in python

The two extra options attributes allow for simple progress bar usage in python algorithms. An example of this being used is in #16240.

**To test:**
1. Edit a python algorithm that takes a while and uses child algorithms.
2. Merge in this branch
3. See that the algorithm still works.
4. Pick some sub-algorithms and add extra attributes `startProgress` and `endProgress`.
5. Run the algorithm again and see that the progress bar now moves

_There was no associated issue._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
